src/functions/module_lookup.ts

```diff
@@ -446,10 +456,11 @@ export const XLOOKUP: AddFunctionDescription = {
     }
 
     const _defaultValue = defaultValue?.();
-    assert(
-      () => !!_defaultValue,
-      _lt("Did not find value '%s' in [[FUNCTION_NAME]] evaluation.", toString(searchKey))
-    );
+    if (!_defaultValue) {
+      throw new NotAvailableError(
```

Comment:
To avoid repetition and maintain consistency

maybe it would be good to create a new helper assert function,

ex ```assertAvailable ``` in which to pass only the condition